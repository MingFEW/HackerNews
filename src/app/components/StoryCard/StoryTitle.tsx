import React from 'react'
import 'twin.macro'

interface StoryTitleProps {
  url: string
  title: string
}

export const StoryTitle: React.FC<StoryTitleProps> = props => {
  const { url, title } = props
  return (
    <a
      href={url}
      target="_blank"
      rel="noreferrer"
      tw="text-base font-bold text-white sm:text-xl hover:opacity-80 hover:no-underline"
    >
      {title}
    </a>
  )
}
