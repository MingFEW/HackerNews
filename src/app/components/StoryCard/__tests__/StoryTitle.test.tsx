import { createRenderer } from 'react-test-renderer/shallow'

import { StoryTitle } from '../StoryTitle'

const shallowRenderer = createRenderer()

describe('<StoryTitle />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<StoryTitle url="https://localhost.com" title="story name" />)
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
