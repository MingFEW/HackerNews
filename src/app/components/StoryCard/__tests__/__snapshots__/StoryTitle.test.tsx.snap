// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryTitle /> should render and match the snapshot 1`] = `
<a
  href="https://localhost.com"
  rel="noreferrer"
  target="_blank"
  tw="text-base font-bold text-white sm:text-xl hover:opacity-80 hover:no-underline"
>
  story name
</a>
`;
